### ISSUE TITLE
Problem with volume on /etc/localtime and SELinux

### ISSUE THREAD

**Issue by YannRobert** [moby/moby#12811]
Hi all

It seems that we cannot map /etc/localtime  as a volume anymore.

When using `-v /etc/localtime:/etc/localtime:ro`
I get the message `Relabeling content in /usr is not allowed.`

```
$ docker --version
Docker version 1.6.0, build 3eac457/1.6.0

$ docker run -ti busybox echo hello
hello

$ docker run -ti -v /etc/localtime:/etc/localtime:ro busybox echo hello
Timestamp: 2015-04-27 13:49:44.183916212 +0000 UTC
Code: System error

Message: Relabeling content in /usr is not allowed.

Frames:

---
0: setupRootfs
Package: github.com/docker/libcontainer
File: rootfs_linux.go@34

---
1: Init
Package: github.com/docker/libcontainer.(*linuxStandardInit)
File: standard_init_linux.go@52

---
2: StartInitialization
Package: github.com/docker/libcontainer.(*LinuxFactory)
File: factory_linux.go@223

---
3: initializer
Package: github.com/docker/docker/daemon/execdriver/native
File: init.go@35

---
4: Init
Package: github.com/docker/docker/pkg/reexec
File: reexec.go@26

---
5: main
Package: main
File: docker.go@29

---
6: main
Package: runtime
File: proc.go@63

---
7: goexit
Package: runtime
File: asm_amd64.s@2232
FATA[0000] Error response from daemon: Cannot start container b1f47bc498357ae6cc086641620999597cbc243c9eee1bab9f81690ed0781072: [8] System error: Relabeling content in /usr is not allowed. 

$ cat /etc/redhat-release
Fedora release 21 (Twenty One)

$ uname -srvmpio
Linux 3.19.5-200.fc21.x86_64 #1 SMP Mon Apr 20 19:51:56 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

$ sestatus
SELinux status:                 enabled
SELinuxfs mount:                /sys/fs/selinux
SELinux root directory:         /etc/selinux
Loaded policy name:             targeted
Current mode:                   enforcing
Mode from config file:          enforcing
Policy MLS status:              enabled
Policy deny_unknown status:     allowed
Max kernel policy version:      29

$ ls -alF /etc/localtime 
lrwxrwxrwx. 1 root root 23 27 avril 12:27 /etc/localtime -> /usr/share/zoneinfo/UTC
```

It was supported on earlier version of Docker.
Am I doing it wrong?

**Comment by cpuguy83** [comment]
ping @rhatdan

**Comment by rhatdan** [comment]
What version of docker are you using

rpm -q docker

**Comment by YannRobert** [comment]
```
$ rpm -q docker-io
docker-io-1.6.0-2.git3eac457.fc21.x86_64
```

**Comment by YannRobert** [comment]
Also noted that mounting the file as read-only volume fails 
```
$ docker run -ti -v /etc/localtime:/etc/localtime:ro busybox echo hello
Timestamp: 2015-04-27 21:34:58.625118144 +0000 UTC
Code: System error

Message: Relabeling content in /usr is not allowed.

Frames:
---
0: setupRootfs
Package: github.com/docker/libcontainer
File: rootfs_linux.go@34
---
1: Init
Package: github.com/docker/libcontainer.(*linuxStandardInit)
File: FATA[0000] Error response from daemon: Cannot start container 35feaf749f688a34571c2e1ac3126e0613a2da428f9b12f6c7cef1d8861261f9: [8] System error: Relabeling content in /usr is not allowed.
```
but mounting it without any specification succeed
```
$ docker run -ti -v /etc/localtime:/etc/localtime busybox echo hello
hello

**Comment by beeradb** [comment]
Not sure if it's related, but I am getting the same error message. My code is at https://github.com/beeradb/webdrupal-docker (see bootstrap.sh and containers/docker-compose.yml)

In my case, it appears to only happen when I try to mount vagrant sync_folders, i.e. /usr/share/webdrupal  mounting into  /usr/share/webdrupal in my container. 

My docker version is "Docker version 1.6.0, build 3eac457/1.6.0". Let me know if I can provide more info.

**Comment by beeradb** [comment]
I should also note it only seems to happen if you include a filemode (which docker-compose does)...

```
[vagrant@localhost ~]$ sudo docker run -ti -v /usr/share/webdrupal:/usr/share/webdrupal busybox echo hello
hello
[vagrant@localhost ~]$ sudo docker run -ti -v /usr/share/webdrupal:/usr/share/webdrupal:ro busybox echo hello
FATA[0000] Error response from daemon: Cannot start container 50122753f10686b779af1a0a674240e9064cb5e082d679ce69b8ee85dc942c9e: [8] System error: Relabeling content in /usr is not allowed. 
[vagrant@localhost ~]$ sudo docker run -ti -v /usr/share/webdrupal:/usr/share/webdrupal:rw busybox echo hello
Timestamp: 2015-04-28 00:06:48.50426352 +0000 UTC
Code: System error

Message: Relabeling content in /usr is not allowed.

Frames:
---
0: setupRootfs
Package: github.com/docker/libcontainer
File: rootfs_linux.go@34
---
1FATA[0000] Error response from daemon: Cannot start container 7db5ba68b77eb216623483196f0f3f202ac6b69a9076f479ad30216c9d81722c: [8] System error: Relabeling content in /usr is not allowed.
```

**Comment by rhatdan** [comment]
Does this work fine with the docker-io package shipped with fedora 21?

**Comment by YannRobert** [comment]
@rhatdan no it does not work precisely with that package, please see my previous comment [https://github.com/docker/docker/issues/12811#issuecomment-96827810] providing the name of the offending package.

You can reproduce in 3 minutes on Digital Ocean by creating a fresh fedora 21 machine and just ```yum install -y docker-io && service docker restart && docker run -ti -v /etc/localtime:/etc/localtime:ro busybox echo hello```

**Comment by rhatdan** [comment]
Yann, I  think I have seen it, please open a Bugzilla on this package.  Working with the latest docker-1.6 packages I am not seeing it, and working with master branch it seems to work fine.

rpm -q docker
docker-1.7.0-2.git50ef691.fc23.x86_64

If you open a bugzilla I will get the packager to push an update to Fedora 21.

**Comment by YannRobert** [comment]
Thank you Dan
I reported the problem in bugzilla as you suggested
https://bugzilla.redhat.com/show_bug.cgi?id=1216151

**Comment by rhatdan** [comment]
Great.  Thanks.

**Comment by ksylvan** [comment]
I'm still seeing this problem with this version:

    $ rpm -q docker-io 
    docker-io-1.6.0-2.git3eac457.fc21.x86_64

And it looks like volume-mounting my host directories don't work either.

**Comment by rhatdan** [comment]
@ksylvan I answered in the bugzillas on the volume mount.  Lets carry on the discussion in the bugzillas.

**Comment by derekwaynecarr** [comment]
I am looking to move the Kubernetes Vagrant cluster to Fedora 21 in order to pick up Docker 1.6+ and I am seeing a similar error when trying to run any pod.

Is there a work-around?

```
Code: System error

Message: Relabeling content in /usr is not allowed.

Frames:
---
0: setupRootfs
Package: github.com/docker/libcontainer
File: rootfs_linux.go@34
---
1: Init
Package: github.com/docker/libcontainer.(*linuxStandardInit)
File: standard_init_linux.go@52
---
2: StartInitialization
Package: github.com/docker/libcontainer.(*LinuxFactory)
File: factory_linux.go@223
---
3: initializer
Package: github.com/docker/docker/daemon/execdriver/native
File: init.go@35
---
4: Init
Package: github.com/docker/docker/pkg/reexec
File: reexec.go@26
---
5: main
Package: main
File: docker.go@29
---
6: main
Package: runtime
File: proc.go@63
---
7: goexit
Package: runtime
File: asm_amd64.s@2232
```

```shell
$ rpm -q docker-io
docker-io-1.6.0-4.git350a636.fc21.x86_64
```

**Comment by thaJeztah** [comment]
@derekwaynecarr according to this comment https://bugzilla.redhat.com/show_bug.cgi?id=1216151#c5 this should be fixed in the docker 1.6.2 packaged by Red Hat.

I'm going to close this issue, because this is an issue in docker as packaged by Red Hat and according to https://bugzilla.redhat.com/show_bug.cgi?id=1216151#c5 this should be resolved in 1.6.2.

Feel free to comment here if you think I closed this by mistake
